### PR TITLE
feat(legacy-preset-chart-big-number): add timestamp above number

### PR DESCRIPTION
**Pull request by krsnik93**
💔 Breaking Changes: No

🏆 Enhancements: add timestamp above the number in BigNumber chart(s)

image

📜 Documentation: NA

🐛 Bug Fix: NA

🏠 Internal: NA

### CODE REVIEW

File: plugins/legacy-preset-chart-big-number/src/BigNumber/controlPanel.tsx [hunk 1]
```diff
@@ -56,6 +56,18 @@ const config: ControlPanelConfig = {
...
+            name: 'show_timestamp',
+            config: {
+              type: 'CheckboxControl',
+              label: t('Show Timestamp'),
+              renderTrigger: true,
+              default: false,
+              description: t('Whether to display the timestamp'),
+            },
+          },
+        ],
```
Comment (villebro) [inline plugins/legacy-preset-chart-big-number/src/BigNumber/controlPanel.tsx:70]: Could we also add a date formatter control that is made visible if `show_timestamp` is `true`? Here's a time formatter control from the ECharts timeseries charts: 
image

**Review by villebro — COMMENTED** [review]
LGTM - let's just add a time formatter to make it possible to customize the time format

File: plugins/legacy-preset-chart-big-number/src/BigNumber/controlPanel.tsx (hunk 1 above)
Comment (krsnik93) [reply plugins/legacy-preset-chart-big-number/src/BigNumber/controlPanel.tsx:76]: added!

**Comment by villebro** [comment]
I'm getting some weird behavior when applying smart date formatting.With adaptive formatter it looks like this:
image

And with some of the other formats it looks like this:
image

To make this behavior more intuitive maybe we could borrow some of the logic from table chart:
https://github.com/apache-superset/superset-ui/blob/1c9baeeb6f03e05ba797452dc463eb8def7996bc/plugins/plugin-chart-table/src/transformProps.ts#L124-L140

**Review by villebro — APPROVED** [review]
Awesome, thanks for quickly iterating on the review comments!

**Comment by junlincc** [comment]
thanks for the improvement! a new control is added?

**Comment by villebro** [comment]
> thanks for the improvement! a new control is added?

Yes - two new ones in fact. One for the checkbox "Show timestamp" (defaults to false), and if that is checked a date format select is displayed to customize the format.

**Pull request merged**